Core/GDCore/Project/Variable.cpp

```diff
@@ -20,25 +20,64 @@ namespace gd {
  * Get value as a double
  */
 double Variable::GetValue() const {
-  if (!isNumber) {
+  if (type == TYPES::NUMBER) {
+    return value;
+  } else if (type == TYPES::STRING) {
     value = str.To<double>();
-    isNumber = true;
+    type = TYPES::NUMBER;
+    return value;
+  } else if (type == TYPES::BOOLEAN) {
+    value = boolVal ? 1.0 : 0.0;
+    type = TYPES::NUMBER;
+    return value;
   }
 
+  // It isn't possible to convert a structural type to a number
+  value = 0.0;
+  type = TYPES::NUMBER;
   return value;
 }
 
 const gd::String& Variable::GetString() const {
-  if (isNumber) {
+  if (type == TYPES::STRING) {
+    return str;
+  } else if (type == TYPES::NUMBER) {
     str = gd::String::From(value);
-    isNumber = false;
+    type = TYPES::STRING;
+    return str;
+  } else if (type == TYPES::BOOLEAN) {
+    str = boolVal ? "1" : "0";
+    type = TYPES::STRING;
+    return str;
   }
 
+  // It isn't possible to convert a structural type to a string
+  str = "0";
+  type = TYPES::STRING;
   return str;

```

Comment:
There was a specific behavior that might be important to keep, but I'm not sure how important it is:

Before, if you set a variable to a number/string, then you somehow set a children (so that it's becoming a structure), then getting the string (transforming it back to a string), you'll get the string (or number) initially set.
While here you'll get "0". 

Seems like a change in behavior. It might be a good idea to keep the fact that the "primitive type" is separated from the "structural type". You can be only one, but if you convert to a structure/array and back to a primitive, we've retained your value (which, again, is not something I would recommend at all and that we could forbid in the future).

Reply:
This is kinda semi working only as restarting the ide (or generally serializing and deserialising) as the serializer only serialized the structure part or the value part. I thought therefore it was unintended behavior.